x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_icon_property.js

```diff
@@ -110,13 +104,24 @@ export class DynamicIconProperty extends DynamicStyleProperty {
     return this._field && this._field.isValid();
   }
 
+  renderLegendDetailRow({ isPointsOnly, isLinesOnly, symbolId }) {
+    return (
+      <BreakedLegend
+        style={this}
+        isPointsOnly={isPointsOnly}
+        isLinesOnly={isLinesOnly}
+        symbolId={symbolId}
+      />
+    );
+  }
+
   renderBreakedLegend({ fieldLabel, isPointsOnly, isLinesOnly }) {
     const categories = [];
     const { stops, fallback } = this._getPaletteStops();
-    stops.map(({ stop, style }) => {
+    stops.map(({ stop, style }, index) => {
       categories.push(
         <Category
-          key={stop}
+          key={index}
```

Comment:
Keep using stop as key. Index is not a good key, https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
will change, but relies on implicit knowledge that the stop-value is derived from a key-map elsewhere in the code.